Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1

```diff
@@ -154,14 +154,17 @@ function InvokeGetRestMethod
     $request = $sonarQubeHostUrl + $query;
     
     if ($useAuth)
-    {
-      
+    {      
        $authHeader = CreateBasicAuthHeaderFromEndpoint
-       $allheaders = @{Authorization = $authHeader}   
-    }  
 
-    $response = Invoke-RestMethod $request -Method Get -Headers $allheaders
+       if (![String]::IsNullOrWhiteSpace($authHeader))
```

Comment:
The actual fix is here - the authHeader can be empty (i.e. anon users). In this case, Powershell's Invoke-RestMethod throws a null ref exception;

Workaround for users: don't use anon connections to SonarQube